beta/nncf/tensorflow/initialization.py

```diff
@@ -0,0 +1,138 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from functools import partial
+from contextlib import contextmanager
+from typing import Dict
+
+import tensorflow as tf
+
+from nncf.common.utils.progress_bar import ProgressBar
+from nncf.common.initialization import NNCFDataLoader
+from nncf.config.structure import BNAdaptationInitArgs
+
+
+class InitializingDataLoader(NNCFDataLoader):
+    """
+    This class wraps the torch.utils.data.DataLoader class,
+    and defines methods to parse the general data loader output to
+    separate the input to the compressed model and the ground truth target
+    for the neural network. This is required for proper initialization of
+    certain compression algorithms.
+    """
+
+    def __init__(self, data_loader):
+        self._data_loader = data_loader
+
+    @property
+    def batch_size(self):
+        for (x, _) in self._data_loader:
+            batch_size = x.shape[0]
+            return batch_size
+
+    def __iter__(self):
+        return iter(self._data_loader)
+
+
+class DataLoaderBNAdaptationRunner:
+    def __init__(self, model, init_device: str, num_bn_forget_steps):
+        self.model = model
+        self.init_device = init_device
+        self.progressbar_description = 'BatchNorm statistics adaptation'

```

Comment:
Do you plan to add the progress bar for the initialization step?

Reply:
Good idea, added